akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileAndResourceDirectives.scala

```diff
@@ -222,7 +222,14 @@ object FileAndResourceDirectives extends FileAndResourceDirectives {
             ResourceFile(url, entry.getSize, entry.getTime)
           }
         } finally jar.close()
-      case _ ⇒ None
+      case _ ⇒
+        val conn = url.openConnection()
```

Comment:
@jypma There's infinite opportunities for this patch to not work. Too many containers out there, too many branches. Even if there was an OSGi test framework for SBT, it's just one of many containers that could depend on custom URLs (eclipse, for example).

The problem is the code itself, it shouldn't be trying to look at opaque URLs. I understand that there's a security hole in directory listings, maybe those should be removed.